tests/components/azure_data_explorer/test_init.py

```diff
@@ -0,0 +1,280 @@
+"""Test the init functions for AEH."""
+from datetime import timedelta
+import logging
+from unittest.mock import patch
+
+from azure.kusto.data.exceptions import KustoAuthenticationError, KustoServiceError
+import pytest
+
+from homeassistant.components import azure_data_explorer
+from homeassistant.components.azure_data_explorer.const import (
+    CONF_SEND_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from .conftest import FilterTest
+from .const import AZURE_DATA_EXPLORER_PATH, BASE_CONFIG_FULL, BASIC_OPTIONS
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
```

Comment:
```suggestion
pytestmark = [
    pytest.mark.freeze_time("2024-01-01 00:00:00"),
]
```

Can we freeze the time to a certain datetime so the tests are deterministic.